migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/DefaultMigrationTarget.java

```diff
@@ -449,19 +449,38 @@ private void saveLibraryDilution(LibraryDilution ldi) throws IOException {
       ldi.setLastModified(timeStamp);
     }
 
+    Collection<ChangeLog> ghostChangeLog = null;
     if (ldi.getLibrary().getId() == AbstractDilution.UNSAVED_ID && ldi.getLibrary().getLibraryAdditionalInfo() != null
         && ldi.getLibrary().getLibraryAdditionalInfo().getPreMigrationId() != null) {
       Long preMigrationId = ldi.getLibrary().getLibraryAdditionalInfo().getPreMigrationId();
+      ghostChangeLog = ldi.getLibrary().getChangeLog();
       ldi.setLibrary(serviceManager.getLibraryDao().getByPreMigrationId(preMigrationId));
       if (ldi.getLibrary() == null) {
         throw new IOException("No Library found with pre-migration ID " + preMigrationId);
       }
     }
     valueTypeLookup.resolveAll(ldi);
     ldi.setId(serviceManager.getDilutionDao().save(ldi));
+    fixDilutionChangeLog(ldi, ghostChangeLog);
     log.debug("Saved library dilution " + friendlyName);
   }
 
+  private void fixDilutionChangeLog(LibraryDilution ldi, Collection<ChangeLog> ghostChangeLog) throws IOException {
+    Library lib = ldi.getLibrary();
+    Collection<ChangeLog> changes = lib.getChangeLog();
+    if (ghostChangeLog != null) {
+      changes.addAll(ghostChangeLog);
+    }
+    String gsleTag = "GSLE" + ldi.getPreMigrationId();
+    String misoTag = ldi.getName() != null ? ldi.getName() : "LDI" + ldi.getId();
+    for (ChangeLog change : changes) {
+      if (change.getSummary().matches("^" + gsleTag + ".*")) {
```

Comment:
I'd feel better if this tag was had a terminating non-numeric character so that `GSLE10` doesn't match `GSLE101`.